src/features/campaigns/components/ActivityList/items/EmailListItem.tsx

```diff
@@ -0,0 +1,68 @@
+import { FC } from 'react';
+import { EmailOutlined, Person } from '@mui/icons-material';
+
+import ActivityListItemWithStats from './ActivityListItemWithStats';
+import useEmail from 'features/emails/hooks/useEmail';
+import useEmailTargets from 'features/emails/hooks/useEmailTargets';
+import ActivityListItem, { STATUS_COLORS } from './ActivityListItem';
+import useEmailState, { EmailState } from 'features/emails/hooks/useEmailState';
+
+interface EmailListItemProps {
+  emailId: number;
+  orgId: number;
+}
+
+const EmailListItem: FC<EmailListItemProps> = ({ orgId, emailId }) => {
+  const { data: email } = useEmail(orgId, emailId);
+  const state = useEmailState(orgId, emailId);
+  const { data: targets } = useEmailTargets(orgId, emailId);
+
+  if (!email) {
+    return null;
+  }
```

Comment:
Minor: Can this early `return` happen before the 2 use* functions?

Reply:
That would make sense, but is actually illegal in React because `use*` functions ("hooks") rely on always being called in the same order (their underlying data in React is store in stacks).

Reply:
🚓

Reply:
Thanks for the teach!